src/circuit/layouter.rs

```diff
@@ -66,57 +67,46 @@ pub trait RegionLayouter<C: Chip>: fmt::Debug {
     ) -> Result<(), Error>;
 }
 
-/// A [`Layouter`] for a single-chip circuit.
-pub struct SingleChip<'a, C: Chip, CS: Assignment<C::Field> + 'a> {
-    cs: &'a mut CS,
-    config: C::Config,
-    loaded: Option<C::Loaded>,
+/// A [`Layouter`] for a single-config circuit.
+pub struct SingleConfigLayouter<'a, F: FieldExt, CS: Assignment<F> + 'a> {
+    /// Constraint system
+    pub cs: &'a mut CS,
     /// Stores the starting row for each region.
     regions: Vec<RegionStart>,
     /// Stores the first empty row for each column.
     columns: HashMap<Column<Any>, usize>,
-    _marker: PhantomData<C>,
+    marker: PhantomData<F>,
```

Comment:
I think that this PhantomData will also now be superfluous; it's implied by the type of `CS`.